### PR TITLE
config.txt/video.md: Add description of Pi4 1366x768 support

### PR DESCRIPTION
**Pull request by 6by9**
List the restriction and the workaround.

### CODE REVIEW

**Comment by 6by9** [comment]
@JamesH65

**Comment by JamesH65** [comment]
@ghollingworth We are planning of adding this note to the docs re: the odd 1366 mode. Does this read OK to you?

@lurch for a second CE view.

File: configuration/config-txt/pi4-hdmi.md [hunk 1]
```diff
@@ -1,38 +1,24 @@
...
 ```
 hdmi_group=2
 hdmi_mode=87
 hdmi_cvt=1360 768 60
 ```
 to config.txt.
 
-Timings specified manually via a ```hdmi_timings=``` line will also need to
-comply with this restriction.
+Timings specified manually via a `hdmi_timings=` line in `config.txt` will also need to comply with the restriction of all timing parameters being divisible by 2.
```
Comment (6by9) [inline configuration/config-txt/pi4-hdmi.md:22]: all *horizontal* timing parameters

**Comment by 6by9** [comment]
We ought to put this as a sticky in the General or Troubleshooting forums too.

**Comment by ghollingworth** [comment]
Looks good to me, makes sense

**Comment by JamesH65** [comment]
OK, if @lurch is happy with the text, I'll merge and add a forum sticky.

**Comment by JamesH65** [comment]
Ah, Lurch on holiday - will do it anyway tomorrow.

**Comment by JamesH65** [comment]
Stickies added.

File: configuration/config-txt/pi4-hdmi.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Raspberry Pi 4 HDMI pipeline
+
+In order to support dual displays, and modes up to 4k60, the Raspberry Pi 4 has updated the HDMI composition pipeline hardware in a number of ways. One of the major changes is that it generates 2 output pixels for every clock cycle.
+
+Every HDMI mode has a list of timings that control all the parameters around sync pulse durations. These are typically defined via a pixel clock, and then a number of active pixels, a front porch, sync pulse, and back porch for each of the horizontal and vertical directions. 
+
+Running everything at 2 pixels per clock means that the Pi4 can not support a timing where any of the horizontal timings are not divisible by 2. The firmware and Linux kernel will filter out any mode that does not fulfill this criteria.
```
Comment (lurch) [inline configuration/config-txt/pi4-hdmi.md:7]: Doesn't seem clear why 1366 uses a horizontal timing that isn't divisible by 2? I guess that doesn't really matter though.

File: configuration/config-txt/pi4-hdmi.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+There is only one mode in the CEA and DMT standard that falls into this category - DMT mode 81, which is 1366x768 @ 60Hz. 1366x768 is already a unusual mode as normally the width of modes is divisible by 8, where 1366 is not.
```
Comment (lurch) [inline configuration/config-txt/pi4-hdmi.md:9]: https://github.com/raspberrypi/documentation/blob/master/configuration/config-txt/video.md also lists DMT mode 86 as 1366x768 @ 60Hz ?

File: configuration/config-txt/pi4-hdmi.md (hunk 2 above)
Comment (JamesH65) [reply configuration/config-txt/pi4-hdmi.md:7]: That what the standard has chosen, no idea why. Timing choice was nothing to do with us.

File: configuration/config-txt/pi4-hdmi.md (hunk 3 above)
Comment (6by9) [reply configuration/config-txt/pi4-hdmi.md:9]: How to explain it to the lay-person? Particularly when the parts of the spec you wish to reference aren't public docs.

I'd referenced front and back porches and sync widths.
Someone (I'm guessing Dom) has posted the full timing table we use at https://pastebin.com/vVLVTMin. It's a direct lift from our firmware source tree.
mode 81
```
{ "1366x768@60", { 81,  1366, HDMI_POSITIVE_SYNC,   70, 143, 213,  768,  HDMI_POSITIVE_SYNC,  3,  3,  24,    0,    0,       0,     60,      HDMI_NONINTERLACED,  85500000},        HDMI_ASPECT_16_9},
```
The 143 and 213 are the problems - they aren't even numbers.
Mode 86 is 1366x768 "reduced blanking"
```
 { "1366x768rb",  { 86,  1366, HDMI_POSITIVE_SYNC,   14,  56,  64,  768,  HDMI_POSITIVE_SYNC,  1,  3,  28,    0,    0,       0,     60,      HDMI_NONINTERLACED,  72000000},        HDMI_ASPECT_16_9},
```
No odd numbers in the horizontal timings, so all good.

The kernel does have this same information at https://github.com/raspberrypi/linux/blob/rpi-5.4.y/drivers/gpu/drm/drm_edid.c#L391 but they use cumulative numbers, so
```
	{ DRM_MODE("1366x768", DRM_MODE_TYPE_DRIVER, 85500, 1366, 1436,
		   1579, 1792, 0, 768, 771, 774, 798, 0,
		   DRM_MODE_FLAG_PHSYNC | DRM_MODE_FLAG_PVSYNC) },
```
The troublesome numbers are still there but less obviously:
1579-1436 = 143
1792-1579 = 213

Feel free to reword, but it's a struggle without getting into huge amounts of technical detail that is likely to confuse or bemuse many.
Possibly just reference that it is the sync width and back porch that are odd values?

The kernel headers do include a pretty ASCII diagram of what the sync periods mean if you want to reference that - https://github.com/raspberrypi/linux/blob/rpi-5.4.y/include/drm/drm_modes.h#L198

File: configuration/config-txt/pi4-hdmi.md (hunk 2 above)
Comment (lurch) [reply configuration/config-txt/pi4-hdmi.md:7]: I just meant that 1366 appears to be divisible by 2, so it's not clear why this violates the rules :wink:  But @6by9 's comment below goes into much more detail...

Comment (6by9) [reply configuration/config-txt/pi4-hdmi.md:7]: It's any of the horizontal timing parameters, hence having described porches etc before.

**Comment by lurch** [comment]
> Particularly when the parts of the spec you wish to reference aren't public docs.

Ah, tricky!

> Mode 86 is 1366x768 "reduced blanking"
No odd numbers in the horizontal timings, so all good.

Does that mean that suggesting DMT mode 86 would also be a viable workaround? Or is it unlikely that any 1366x768 displays actually support the "reduced blanking" mode?

**Comment by 6by9** [comment]
> Does that mean that suggesting DMT mode 86 would also be a viable workaround? Or is it unlikely that any 1366x768 displays actually support the "reduced blanking" mode?

We've been down that route.
There is no **guarantee** that the display supports 1366x768rb. The EDID in theory lists all the modes that are supported, but they'll often support others. How do you know if they do? You don't, therefore you can't rely on it and automatically make the substitution.

**Comment by lurch** [comment]
Thanks for the clarifications Dave, please accept my apologies for my nosiness :wink:

**Comment by 6by9** [comment]
No problem. If I can clarify it for you then potentially you can put it into words for others. I'm often too close to the issue to express it clearly.

**Comment by 6by9** [comment]
Ooh, potentially the DMT spec is public
https://glenwing.github.io/docs/VESA-DMT-1.12.pdf page 56 for mode 81 (0x51)
I know that website did have a DMCA takedown over some of the HDMI specs, but this one is still up.

**Pull request merged**